ghcjs-src/Miso/FFI.hs

```diff
@@ -101,23 +104,41 @@ foreign import javascript unsafe "console.log($1);"
 foreign import javascript unsafe "$r = JSON.stringify($1);"
   stringify' :: JSVal -> IO JSString
 
-foreign import javascript unsafe "$r = JSON.parse($1);"
+-- | Parse a raw string into JSON. Accepts any JS value.
+foreign import javascript unsafe "$r = JSON.parse('' + $1);"
   parse' :: JSVal -> IO JSVal
 
 -- | Converts a JS object into a JSON string
 stringify :: ToJSON json => json -> IO JSString
 {-# INLINE stringify #-}
 stringify j = stringify' =<< toJSVal (toJSON j)
 
--- | Parses a JSString
+-- | Parses a JSVal (which of course should be a string). Less safe
+-- | than 'parseString'.
+parseString :: FromJSON json => JSString -> IO json
+{-# INLINE parseString #-}
+parseString = parse . unsafeCoerce
+
+-- | Parses a JSVal (which of course should be a string). Less safe
+-- | than 'parseString'.
```

Comment:
This should reference to `safeParseString`. I would also suggest to remove the `safe` prefix and instead add an `unsafe` prefix to this function. The docs should also mention in what way it is unsafe.

Reply:
Totally in favor of that; I just didn't want to rename a function which is used throughout the code. Can do though.